test/compress/collapse_vars.js

```diff
@@ -1047,10 +1046,9 @@ collapse_vars_object: {
     }
     expect: {
         function f0(x, y) {
-            var z = x + y;
             return {
                 get b() { return 7; },
-                r: z
+                r: x + y
```

Comment:
What in collapse_vars prevented this particular optimization before? I don't see any side effects above it.